### PR TITLE
chore(http): delete "Basics of HTTP" page in favor of HTTP landing page, reorganize index

### LINKED ISSUE #4922: Unable to access documentation for "Frame and message structure in HTTP/2"
Hi Team,

First of all, thank you for your efforts in creating the documentation for HTTP, I was planning to learn more about HTTP2.0 implementation and came across an issue due to which I am not able to access the page. The below URL which is shared on other pages(I got redirected from https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP) as well

https://developer.mozilla.org/en-US/docs/Web/HTTP/Frame_and_message_structure_in_HTTP_2 

Error details:

Page not found
Sorry, the page /en-US/docs/Web/HTTP/Frame_and_message_structure_in_HTTP_2 could not be found.

Thanks and Regards,
Ujjwal Sharma

### LINKED ISSUE #6955: Issue with "Basics of HTTP": duplicative with root HTTP page
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP

#### What information was incorrect, unhelpful, or incomplete?

This page duplicates a lot of the links in https://developer.mozilla.org/en-US/docs/Web/HTTP -- can its content be consolidated there? happy to do a PR.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/http/basics_of_http`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/http/basics_of_http/index.html
* Last commit: https://github.com/mdn/content/commit/4062cfb18880e12a6f892d794114c9c2f062a290
* Document last modified: 2021-04-19T04:01:15.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
### Description

This PR removes the "Basics of HTTP" document in favor of using the https://developer.mozilla.org/en-US/docs/Web/HTTP landing page to list everything.

### Motivation

There are guides that are less visible nested further down the tree, this surfaces those resources, and reorganizes the index for maintainability.

__dependencies:__

Yari sidebars need update to reflect removal: 

- [x] https://github.com/mdn/yari/pull/11753 

Fixes https://github.com/mdn/content/issues/6955
Fixes https://github.com/mdn/content/issues/4922

### CODE REVIEW

**Comment by bsmth** [comment]
> @bsmth can we get this merged despite the Yari PR? We need to merge this beforehand anyway.

Thanks for asking, coming back to it again, I believe it's safe to merge before the sidebars changes. The only oddity would be `HTTP/Basics_of_HTTP` in the sidebar redirecting to `/HTTP` while the Yari PR is in-flight.

File: files/en-us/web/http/index.md [hunk 1]
```diff
@@ -6,46 +6,115 @@ page-type: landing-page
...
+  - : HTTP/1.1 provides a mechanism to upgrade an already-established connection to a different protocol using the {{HTTPHeader("Upgrade")}} header.
+    A client can upgrade a connection from HTTP/1.1 to HTTP/2, or an HTTP(S) connection to a [WebSocket](/en-US/docs/Web/API/WebSocket) (`ws` / `wss`).
+- [Evolution of HTTP](/en-US/docs/Web/HTTP/Evolution_of_HTTP)
+  - : HTTP was created in the early 1990s and has been extended several times.
+    This article goes through its history and describes HTTP/0.9, HTTP/1.0, HTTP/1.1, through HTTP/2 and HTTP/3, as well as novelties introduced over the years.
+
+## Practical and topical guides
 
-## Reference
+Learn about common communication patterns and methods for improving security, resource usage and network efficiency.
```
Comment (Josh-Cena) [inline files/en-us/web/http/index.md:68]: FWIW, I don't really get what the difference between these and "concepts" are. For example "Content negotiation" is as guide-y as "HTTP caching" is to me.

**Review by Josh-Cena — APPROVED** [review]
I've always been confused about the HTTP doc structure, but approving so you can merge any time. But I think anything that's not listed in "references" and "security&privacy" sections actually looks like a "guide".

File: files/en-us/web/http/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/http/index.md:68]: That's true, what helped me categorize them here at first is that the guides in the 'practical' section here are more 'how-to' (how to do caching, redirection), whereas above, I saw these as describing concepts that are generic. Although, one section "Guides" may be more appropriate with the `<dt>` saying what to expect from the content:

> This guide explains how to...

Comment (Josh-Cena) [reply files/en-us/web/http/index.md:68]: Yes, that would be my thought too. Every guide article should ideally be half concept-explaining and half usecase walkthrough.

Comment (hamishwillee) [reply files/en-us/web/http/index.md:68]: So the change here is that previously all the guide was grouped under tutorials and now you have split that into concepts and practical guides?

I think this is better that what was there and I don't have a better suggestion.

**Review by hamishwillee — APPROVED** [review]
@bsmth Does this need a review from me?

File: files/en-us/web/http/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/http/index.md]: >previously all the guide was grouped under tutorials and now you have split that into concepts and practical guides

There was an HTTP guide nested under "[Basics of HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP)", this PR brings these sub-pages into the index so the content is visible within the overall technology category. I think there's some opportunity to clean up / remove redundant info in future, but this is step one to tidying up.

>concepts and practical guides

That was my first thought, but I'm not sure about making this separation, per Josh' comment.

**Comment by bsmth** [comment]
> @bsmth Does this need a review from me?

Thanks for checking, I think this is in a good state now, happy to hear your thoughts on [the landing page](https://pr35774.content.dev.mdn.mozit.cloud/en-US/docs/Web/HTTP)

**Review by Josh-Cena — APPROVED** [review]
This looks great!

**Comment by hamishwillee** [comment]
Thank you. Merging, as "much better".

**Comment by bsmth** [comment]
Thanks, both! Yari PR has been merged, so the sidebar changes will be landing soon.

**Pull request merged**